client/src/pages/SignUp.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from 'react';
+import Joi from 'joi';
+
+import {
+  Avatar,
+  Button,
+  CssBaseline,
+  TextField,
+  Link,
+  Grid,
+  Box,
+  Typography,
+  Container,
+} from '@mui/material';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+import axios from 'axios';
+import BasicSelect from '../components';
+
+const Copyright = (props: any) => {
+  return (
+    <Typography
+      variant="body2"
+      color="text.secondary"
+      align="center"
+      {...props}
+    >
+      {'Copyright © '}
+      <Link color="inherit" href="/">
+        Menu Mingle
+      </Link>
+      {new Date().getFullYear()}.
+    </Typography>
+  );
+};
+
+const serverUrl = import.meta.env.VITE_APP_SERVER_URL;
+
+const SignUp = () => {
+  const [role, setRole] = React.useState();
+  const [errors, setErrors] = React.useState({});
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const data = new FormData(event.currentTarget);
+
+    const schema = Joi.object({
```

Comment:
Schemas should be in their own file in the utils directory. Readability is essential here.

Reply:
done @moustf   ❤️